crates/evm/src/types.rs

```diff
@@ -0,0 +1,51 @@
+//! Includes the types used by the citrea-evm.
+use std::collections::BTreeMap;
+
+use reth_primitives::{Address, B256};
+use serde::{Deserialize, Serialize};
+
+/// BlockOverrides is a set of header fields to override.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(default, rename_all = "camelCase", deny_unknown_fields)]
+pub struct BlockOverrides {
+    /// Overrides the block number.
+    ///
+    /// For `eth_callMany` this will be the block number of the first simulated block. Each
+    /// following block increments its block number by 1
+    // Note: geth uses `number`, erigon uses `blockNumber`
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        alias = "blockNumber"
+    )]
+    pub number: Option<u64>,
```

Comment:
one problem with using u64's here is that we'll have a hard time with how all eth jsonrpc clients are implemented. 

for things like number, all rpc args are normally hex, and u64 will have a problem with that.

alloy U64 has out-of-box ethereum rpc support take a look at that pls